pkg/localstorev2/internal/upload/uploadstore.go

```diff
@@ -327,7 +340,25 @@ func (u *uploadPutter) Put(ctx context.Context, chunk swarm.Chunk) error {
 		return errPutterAlreadyClosed
 	}
 
-	u.split++
+	switch item, loaded, err := stampindex.LoadOrStore(u.s, stampIndexUploadNamespace, chunk); {
+	case err != nil:
+		return fmt.Errorf("load or store stamp index for chunk %v has fail: %w", chunk, err)
+	case !loaded:
+		u.split++
+	case loaded && item.ChunkIsImmutable:
+		return errOverwriteOfImmutableBatch
+	case loaded && !item.ChunkIsImmutable:
+		prev := binary.BigEndian.Uint64(item.BatchTimestamp)
+		curr := binary.BigEndian.Uint64(chunk.Stamp().Timestamp())
+		if prev >= curr {
+			return errOverwriteOfNewerBatch
+		}
+		if err := stampindex.Store(u.s, stampIndexUploadNamespace, chunk); err != nil {
```

Comment:
Ideally, we should delete the old chunk here. But due to the new design, the chunk will anyway be deleted after pushing. So we can just leave it as is and allow this new push to overwrite it in the network.

Reply:
Done.